UnityProject/Assets/Scripts/Lighting System/TextureDataRequest.cs

```diff
@@ -1,22 +1,29 @@
-﻿using Unity.Collections;
+﻿using System.Collections.Generic;
+using Unity.Collections;
+using Unity.Jobs;
 using UnityEngine;
 using UnityEngine.Rendering;
-using System.Collections.Generic;
 
 public class TextureDataRequest
 {
 	private bool iRequestLoaded;
-	private List<Color32> textureBuffer = new List<Color32>();
 	private int height;
 	private int width;
+	NativeArray<Color32> colors;
+
+	public void DeallocateOnClose()
```

Comment:
Would editor errors go away if you do it like this:
```suggestion
	public void TryDispose()
	{
	if (colors?.Length > 0)
			{
				colors.Dispose();
			}
	}
```
And also use it inside StoreData?

Reply:
Nay, this is a memory leak error so the NativeArray has been allocated.

Reply:
And looks like it only happens if you play in the editor. Stop and then recompile the script. Weird